BLUESPAWN-win-client/src/util/filesystem/FileSystem.cpp

```diff
@@ -188,18 +187,19 @@ namespace FileSystem {
             IO_STATUS_BLOCK IoStatus{};
             InitializeObjectAttributes(&attributes, &UnicodeName, OBJ_KERNEL_HANDLE | OBJ_CASE_INSENSITIVE, nullptr,
                                        nullptr);
-            NTSTATUS Status{ Linker::NtCreateFile(&hFile, GENERIC_READ | GENERIC_WRITE, &attributes, &IoStatus, nullptr,
-                                                  FILE_ATTRIBUTE_NORMAL, FILE_SHARE_READ, FILE_OPEN,
-                                                  FILE_SEQUENTIAL_ONLY, nullptr, 0) };
+            NTSTATUS Status{ Linker::NtCreateFile(
+                &hFile, GENERIC_READ | GENERIC_WRITE, &attributes, &IoStatus, nullptr, FILE_ATTRIBUTE_NORMAL,
+                FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, FILE_OPEN, FILE_SEQUENTIAL_ONLY, nullptr, 0) };
             if(NT_SUCCESS(Status)) {
                 this->hFile = hFile;
                 bFileExists = true;
                 bWriteAccess = true;
                 bReadAccess = true;
             } else if(Status == 0xC0000022 || Status == 0xC0000043) {   //STATUS_ACCESS_DENIED, STATUS_SHARING_VIOLATION
                 Status = Linker::NtCreateFile(&hFile, GENERIC_READ, &attributes, &IoStatus, nullptr,
-                                              FILE_ATTRIBUTE_NORMAL, FILE_SHARE_READ, FILE_OPEN, FILE_SEQUENTIAL_ONLY,
-                                              nullptr, 0);
+                                              FILE_ATTRIBUTE_NORMAL,
+                                              FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE, FILE_OPEN,

```

Comment:
so this if branch is triggered in the event that we get some sort of ACCESS_DENIED from the first attempt to open the file. Are you sure we want to again try `FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE` instead of just `FILE_SHARE_READ` here during this second attempt?

Reply:
Yes, that is what we want. If I don't specify FILE_SHARE_WRITE and the handle is open for write somewhere else, it'll fail. This will enable us to open the file for read.